tools/vulnerability-scan/wss-scan.sh

```diff
@@ -47,33 +47,70 @@ basepath=$baseDirPath"/repos"
 
 echo "Cleaning up scan directories if already present"
 rm -rf $basepath
+echo "Cleaning up temp file that will affect scan"
+rm -rf /tmp/ws*
 
 mkdir -p $basepath
  
 # clone the desired Repos for scanning 
 for repo in $gitRepos
 do
   echo "Cloning repo "$gitBasePath$repo
-  git clone "$gitBasePath$repo".git $basepath"/"$repo
+  if [ ${repo} == "OpenSearch" ]
+  then
+    echo "Cloning "$repo" at branch 1.x"
+    git clone -b 1.x "$gitBasePath$repo".git ${basepath}/${repo}
+    rm -rf ${basepath}/${repo}/qa
+  else
+    echo "Cloning "$repo" at main branch" 
+    git clone "$gitBasePath$repo".git ${basepath}/${repo}
+  fi
 done
 
 echo -n > info.txt
 target='OpenSearch'
-mvn -v; node -v; npm -v; yarn -v
+gradle -v; mvn -v; node -v; npm -v; yarn -v
 
 # scan the Repos using the WhiteSource Unified Agent
 for repo in $gitRepos
 do
   repo_path=$basepath"/"$repo
   if [ -d "$repo_path" ]
   then
+    cd $repo_path
     echo "Scanning repo: "$gitBasePath$repo " Project: " $repo
-    java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product "$target" -project $repo
+# Generate temporary `settings.gradle` file based on the name in `build.gradle`    
+    if [ -e "build.gradle" ]
+    then
+        echo "build.gradle for $repo exist in root"
+        if [ ! -e "settings.gradle" ]
+        then
+            echo "settings.gradle not exist in $repo, create one based on build.gradle name"
+            settings_gradle_content=`cat build.gradle | grep name | grep -i $repo | head -n 1 | sed 's/name/rootProject.name =/g' | awk '{$1=$1};1'`
+            echo $settings_gradle_content > settings.gradle
+            cat settings.gradle
+        else
+            echo "settings.gradle exist $repo"
+        fi
+    else
+# Exceptions for some repos that have `build.gradle` in sub-folder
+        if [ ${repo} == "observability" ]
+        then
+          cd opensearch-observability
+          settings_gradle_content=`cat build.gradle | grep name | grep -i $repo | head -n 1 | sed 's/name/rootProject.name =/g' | awk '{$1=$1};1'`
+          echo $settings_gradle_content > settings.gradle
+          cat settings.gradle
```

Comment:
Why is there only observability repo?
SQL, Reports all with this exception.

Plus, consider put this section into a function, as you are rewriting them multiple times.

Reply:
After testing, dashboards reports contains `[settings.gradle](https://github.com/opensearch-project/dashboards-reports/blob/main/reports-scheduler/settings.gradle)` in its sub-folder, so WhiteSource can successfully track the file location with no needs to go in there. Same for SQL. 

The essence of having WhiteSource running for gradle project is to have the `settings.gradle` present. Otherwise it may look into the `settings.gradle` in build repo in our case and cause the incomplete scan.